bazel: support ignoring logic test files according to a glob

**Describe the problem**

Stray files can panic in the generation code:

```
/data/home/kena/.cache/bazel/_bazel_kena/9b9f53b7334495d6a871cbb472c466bb/execroot/com_github_cockroachdb_cockroach/bazel-out/freebsd-fastbuild/bin/pkg/cmd/generate-logictest/generate-logictest_/generate-logictest.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/active_version~: u
nknown config name local-v1.1@v1.0-noupgrade
panic: /data/home/kena/.cache/bazel/_bazel_kena/9b9f53b7334495d6a871cbb472c466bb/execroot/com_github_cockroachdb_cockroach/bazel-out/freebsd-fastbuild/bin/pkg/cmd/generate-logictest/generate-logictest_/generate-logictest.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/active_vers
ion~: unknown config name local-v1.1@v1.0-noupgrade

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase.stdlogger.Fatalf({}, {0x3f9977?, 0xc311e7?}, {0xc000542120?, 0x271100?, 0x1?})
        github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase/logictestbase.go:570 +0x8b
github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase.processConfigs({0x5b6630, 0x1653168}, {0xc000536c80, 0x135}, {0xc0004aa810, 0x5, 0x153?}, {0xc0005131f0, 0x1, 0x1})
        github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase/logictestbase.go:697 +0x4c5
github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase.ReadTestFileConfigs({0x5b6630, 0x1653168}, {0xc000536c80, 0x135}, {0xc0004aa810, 0x5, 0x5})
        github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase/logictestbase.go:612 +0x395
github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase.ConfigCalculator.Enumerate({{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0}, {0xc00073fd08, 0x1, 0x1})
        github.com/cockroachdb/cockroach/pkg/sql/logictest/logictestbase/logictestbase.go:821 +0x45e
main.(*testdir).addLogicTests(0xc000516120, {0x3e63c8, 0x9}, {{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0})
```

Ideally gitignored files would be ignored by the generator or a custom ignore glob could be supplied.

Jira issue: CRDB-18292

--- a/pkg/cloud/externalconn/connectionpb/BUILD.bazel
+++ b/pkg/cloud/externalconn/connectionpb/BUILD.bazel
@@ -8,10 +8,7 @@ proto_library(
     srcs = ["connection.proto"],
     strip_import_prefix = "/pkg",
     visibility = ["//visibility:public"],
-    deps = [
-        "//pkg/cloud/cloudpb:cloudpb_proto",
-        "@com_github_gogo_protobuf//gogoproto:gogo_proto",
-    ],
+    deps = ["@com_github_gogo_protobuf//gogoproto:gogo_proto"],
 )
 
 go_proto_library(
@@ -20,10 +17,7 @@ go_proto_library(
     importpath = "github.com/cockroachdb/cockroach/pkg/cloud/externalconn/connectionpb",
     proto = ":connectionpb_proto",
     visibility = ["//visibility:public"],
-    deps = [
-        "//pkg/cloud/cloudpb",
-        "@com_github_gogo_protobuf//gogoproto",
-    ],
+    deps = ["@com_github_gogo_protobuf//gogoproto"],
 )
 
 go_library(

--- a/pkg/cloud/externalconn/connectionpb/connection.proto
+++ b/pkg/cloud/externalconn/connectionpb/connection.proto
@@ -13,7 +13,6 @@ package cockroach.cloud.externalconn.connectionpb;
 option go_package = "connectionpb";
 
 import "gogoproto/gogo.proto";
-import "cloud/cloudpb/external_storage.proto";
 
 // ConnectionType is the type of the External Connection object.
 enum ConnectionType {

